### PR TITLE
update to modern GHC

### PR DESCRIPTION
**Pull request by cblp**
and fix some warnings

### CODE REVIEW

File: src/Network/Email/Header/Render.hs [hunk 1]
```diff
@@ -48,7 +48,6 @@ import qualified Data.ByteString              as B
 import qualified Data.ByteString.Lazy.Builder as B
 import           Data.CaseInsensitive         (CI)
 import qualified Data.CaseInsensitive         as CI
-import           Data.Monoid
```
Comment (knrafto) [inline src/Network/Email/Header/Render.hs]: It looks like this is causing the tests to fail: 

```
src/Network/Email/Header/Render.hs:146:44:
    Not in scope: `mempty'
```

**Comment by knrafto** [comment]
Thanks a lot! As you can tell I haven't touched this in a while. It looks like Travis can't compile it though - does it work for you?

**Comment by cblp** [comment]
Yes, tests are passed for me locally. I use `stack test`. We can add stack support travis using http://docs.haskellstack.org/en/stable/travis_ci/

**Comment by cblp** [comment]
Ah, Travis uses GHC installed in Ubuntu, so it is old a little. The best solution would be to test on different GHC versions, I think.

**Comment by cblp** [comment]
Should I try to add stack support in this branch?

**Comment by knrafto** [comment]
Specifying base >= 4.8 in the cabal file might be sufficient, since that's when Data.Monoid was added to Prelude. I can't push to your branch, so could you try it out?

**Comment by cblp** [comment]
Done. I also increased version, because dropping support for base < 4.8 may break something.

**Comment by knrafto** [comment]
Hmm, it's still failing. The tests pass locally for me too, so I'll merge this and try to figure out Travis later. Thanks for the PR!

**Comment by cblp** [comment]
Ubuntu already has its system `base` package, and cabal-install can't install another version.

**Comment by cblp** [comment]
Could you please release this to Hackage?

**Comment by knrafto** [comment]
Yeah, I'll get to it later today!

**Comment by knrafto** [comment]
Uploaded: http://hackage.haskell.org/package/email-header-0.4.0

**Comment by Gabriel439** [comment]
You can support older versions of `base` without incurring unused import warnings for newer versions of `base` by using the following `CPP`:

```haskell
#if MIN_VERSION_base(4,8,0)
#else
import Data.Monoid
#endif
```

**Pull request merged**